lib/publiccloud/instance.pm

```diff
@@ -919,9 +919,18 @@ sub do_systemd_analyze_time {
 
 sub upload_supportconfig_log {
     my ($self, %args) = @_;
-    $self->ssh_script_run(cmd => 'sudo supportconfig -R /var/tmp -B supportconfig -x AUDIT', timeout => 600);
-    $self->ssh_script_run(cmd => 'sudo chmod 0644 /var/tmp/scc_supportconfig.txz');
-    $self->upload_log('/var/tmp/scc_supportconfig.txz', failok => 1, timeout => 600);
+    return unless (get_var('SUPPORTCONFIG_LOG', 1));
+    # to exclude nothings, set "-"
+    my $exclude = get_var('SUPPORTCONFIG_EXCLUDE', 'AUDIT');
+    $exclude = ($exclude =~ s/[-\s]+//gr) ? '-x ' . $exclude : '';
+    my $log = "/var/tmp/scc_supportconfig";
+    my $cmd = "timeout --preserve-status 900 sudo supportconfig -R " . dirname($log) . " -B supportconfig $exclude";
+    record_info('supportconfig start', "start time: " . localtime());
+    my $err = $self->ssh_script_run(cmd => "$cmd", timeout => 960, proceed_on_failure => 1);
+    $self->ssh_script_run(cmd => "ls -l " . dirname($log) . "; [ -f $log.txz ] || sudo tar -cv -f $log.txz -J $log; sudo chmod 0644 $log.txz", proceed_on_failure => 1);
```

Comment:
can you elaborate this line ? AFAIK there is just single arch already compressed by supportconfig tool itself , so why do we need to grep all files and compress them again ?

Reply:
When supportconfig executes the planned sequence of "features", collecting related logs, a specific directory is filled with all the results. At the moment that a feature goes in hang all those results are available and very useful to investigate as the failed test issue as the hanging feature, but lost if not collected.

Reply:
IMHO this point is much relevant and should not be removed from eventual PR merging

Reply:
so this is needed ONLy when supportconfig hangs , but currently we ALWAYS executing it and creating tarball even when we already have a tarball . this command make sense only `if($err)` otherwise we can just copy normal archive created by the tool

Reply:
ok I see . I misread this oneliner . IMHO it is better to keep more code here but make it more readable , but I will not insist on this . Feel free to keep but please leave some comment explaining what is going on ...